app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,12 @@
+module Archivable
+  extend ActiveSupport::Concern
+
+  included do
+    scope :archived, -> { where('archived_at is not null') }
+    scope :published, -> { where(archived_at: nil) }
+  end
+
+  def archive
+    update_attributes(archived_at: Time.now)
```

Comment:
Ah, I didn't realize `update_attribute` isn't threadsafe, thanks! I guess most of the places you'd use it (updating boolean flags), it wouldn't matter, but here it slightly could.